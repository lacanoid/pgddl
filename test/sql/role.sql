\pset null _null_
\pset format unaligned

begin;

create user pgddl_test_user nologin nosuperuser;
--alter user pgddl_test_user with password 'md5db6d137a5ac4e8c22bed58e3e6687aca';

alter user pgddl_test_user valid until '2100-01-01';

comment on role pgddl_test_user
is 'pg_ddl test user...';

alter user pgddl_test_user 
set standard_conforming_strings=true;

create role pgddl_test_user2 login connection limit 100;
create role pgddl_test_user3 valid until 'infinity';

grant pgddl_test_user2 to pgddl_test_user;
grant pgddl_test_user3 to pgddl_test_user with admin option;

select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user';

select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user3';

select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user2';

abort;

