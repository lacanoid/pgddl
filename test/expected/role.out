\pset null _null_
\pset format unaligned
begin;
create user pgddl_test_user nologin nosuperuser;
--alter user pgddl_test_user with password 'md5db6d137a5ac4e8c22bed58e3e6687aca';
alter user pgddl_test_user valid until '2100-01-01';
comment on role pgddl_test_user
is 'pg_ddl test user...';
alter user pgddl_test_user 
set standard_conforming_strings=true;
create role pgddl_test_user2 login connection limit 100;
create role pgddl_test_user3 valid until 'infinity';
grant pgddl_test_user2 to pgddl_test_user;
grant pgddl_test_user3 to pgddl_test_user with admin option;
select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user';
ddlx_create
CREATE GROUP pgddl_test_user;

ALTER ROLE pgddl_test_user WITH
  NOLOGIN
  NOSUPERUSER
  INHERIT
  NOCREATEDB
  NOCREATEROLE
  NOREPLICATION;

COMMENT ON ROLE pgddl_test_user IS 'pg_ddl test user...';
ALTER ROLE pgddl_test_user VALID UNTIL '2100-01-01 00:00:00-08';

ALTER ROLE pgddl_test_user SET standard_conforming_strings=true;

GRANT pgddl_test_user2 TO pgddl_test_user;
GRANT pgddl_test_user3 TO pgddl_test_user WITH ADMIN OPTION;


(1 row)
select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user3';
ddlx_create
CREATE GROUP pgddl_test_user3;

ALTER ROLE pgddl_test_user3 WITH
  NOLOGIN
  NOSUPERUSER
  INHERIT
  NOCREATEDB
  NOCREATEROLE
  NOREPLICATION;

ALTER ROLE pgddl_test_user3 VALID UNTIL 'infinity';

GRANT pgddl_test_user3 TO pgddl_test_user WITH ADMIN OPTION;


(1 row)
select ddlx_create(oid) from pg_roles where rolname='pgddl_test_user2';
ddlx_create
CREATE USER pgddl_test_user2;

ALTER ROLE pgddl_test_user2 WITH
  LOGIN
  NOSUPERUSER
  INHERIT
  NOCREATEDB
  NOCREATEROLE
  NOREPLICATION;

ALTER ROLE pgddl_test_user2 CONNECTION LIMIT 100;

GRANT pgddl_test_user2 TO pgddl_test_user;


(1 row)
abort;
